From 35bba88e75311242841dde83073c211310ec0259 Mon Sep 17 00:00:00 2001
From: Matthew Hagan <mnhagan88@gmail.com>
Date: Wed, 4 May 2022 08:07:06 +0100
Subject: [PATCH] bcm53xx: partition fixups for Meraki MX64/MX65

We need to enlarge the u-boot partition to add extra features like UBI
booting. The shmoo and newly created env partitions can easily be moved
to the nvram partition. This fixup allows u-boot to be enlarged to up to
1MiB.

Signed-off-by: Matthew Hagan <mnhagan88@gmail.com>
---
 arch/arm/boot/dts/broadcom/bcm958625-meraki-mx6x-common.dtsi | 20 ++++++++++++--------
 1 file changed, 12 insertions(+), 8 deletions(-)

--- a/arch/arm/boot/dts/broadcom/bcm958625-meraki-mx6x-common.dtsi
+++ b/arch/arm/boot/dts/broadcom/bcm958625-meraki-mx6x-common.dtsi
@@ -84,13 +84,7 @@
 
 		partition@0 {
 			label = "u-boot";
-			reg = <0x0 0x80000>;
-			read-only;
-		};
-
-		partition@80000 {
-			label = "shmoo";
-			reg = <0x80000 0x80000>;
+			reg = <0x0 0x100000>;
 			read-only;
 		};
 
@@ -101,7 +95,18 @@
 
 		partition@400000 {
 			label = "nvram";
-			reg = <0x400000 0x100000>;
+			reg = <0x400000 0x40000>;
+		};
+
+		partition@440000 {
+			label = "u-boot-env";
+			reg = <0x440000 0x40000>;
+		};
+
+		partition@480000 {
+			label = "shmoo";
+			reg = <0x480000 0x80000>;
+			read-only;
 		};
 
 		partition@500000 {
